services/apiService.ts

```diff
@@ -342,15 +342,20 @@ export const getUniqueVisitorCount = async (id: number) => {
   }
 };
 
-export async function getQuestById(questId: string | number) {
-  const response = await fetch(`${baseurl}/get_quest?id=${questId}`);
-  if(!response.ok) {
-    return await response.text()
+export async function getQuestById(id: string) {
+  const response = await fetch(`${baseurl}/get_quest?id=${id}`);
+  if (!response.ok) {
+    throw new Error(`Error fetching quest data: ${response.status}`);
+  }
+  try {
+    const data: QuestDocument = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error parsing quest data:", error);
+    throw new Error("Failed to parse quest data");
```

Comment:
no need to throw error, just returning a empty value should do

Reply:
ok chief